haystack/core/pipeline/base.py

```diff
@@ -929,7 +914,8 @@ def _distribute_output(
                         run_queue.remove(pair)
                     if pair in waiting_queue:
                         waiting_queue.remove(pair)
-                    run_queue.append(pair)
+                    # run_queue.append(pair)
+                    run_queue.insert(0, pair)
```

Comment:
Changing that doesn't affect any of the existing behavioural tests, though I'd lean onto this change. We always state that Components that are greedy and variadic run as soon as they receive input so this would be the correct logic in my opinion. Also this would be the time to make this kind of change I think.

> If so, have we also updated other parts of the logic that makes assumptions about the LIFO/LILO nature of the queue?

This is the only place in which we don't append.